### DISCUSSION TITLE
cdn delivery

### DISCUSSION THREAD

**Discussion by jimmywarting** [npm/feedback#745] [Registry]
now that nodejs is implementing http imports. And that Deno an browser already have ESM module http imports
what would you think about enabling cdn usages for those packages that is written as ESM module and has `{"type": "module"}` in package json

Something like:
```js
import foo from 'https://cdn.npmjs.com/foo@3.1.x/index.js'
```
There is no reason why it could not have version control in the url either

**Comment by ljharb** [comment]
npm has nothing to do with this - what you’re describing is how node’s (very insecure) https imports will already work. npm only reads package.json and installs node_modules to disk.

If I’m misunderstanding and you’re asking for npm to basically implement unpkg.com, then I’d imagine that’s way beyond the scope of an RFC.

**Reply by thescientist13** [reply]
> asking for npm to basically implement unpkg.com

that was my impressions of this request, and also share your conclusion of that suggestion as well.  The registry and client probably keep the team busy enough as it is.  😅 

Otherwise, as you say resolution of _node_modules_ is a Node function, not an npm (client) function.

**Reply by jimmywarting** [reply]
> asking for npm to basically implement unpkg.com

that was the request...

> what you’re describing is how node’s (very insecure) https imports will already work

how would it be any less secure from running a direct import from static CDN with version control directly from npm?

**Reply by ljharb** [reply]
It's not; those are both incredibly insecure, because DNS is involved.

**Reply by jimmywarting** [reply]
DNS is also involved when you run npm install to fetch archived packages to node_modules folder from npm

**Reply by ljharb** [reply]
Very true! However, that happens at build time, long before production, and since the best practice has that always connecting to an internal registry, and registries provide checksums (webservers do not), it's much, much more secure.
